src/user/dto/searchUsers.dto.ts

```diff
@@ -0,0 +1,19 @@
+import {IsOptional, MinLength, IsPhoneNumber, IsString} from 'class-validator';
+
+export class SearchUsersDto {
+    @IsString()
+    @IsOptional()
+    ids?: [string];
+
+    @MinLength(6)
+    @IsOptional()
+    login?: string;
+
+    @IsPhoneNumber()
+    @IsOptional()
+    phone?: string;
+
+    @IsString()
+    @IsOptional()
+    role?: string;
```

Comment:
Consider enumerating roles to check those list instead of random string. Especially, when validating incoming requests